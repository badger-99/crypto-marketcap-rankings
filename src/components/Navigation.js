import { NavLink } from 'react-router-dom';

const Navigation = () => (
  <nav>
    <ul>
      <li>
        <NavLink className="link" to="/1">#1</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/2">#2</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/3">#3</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/4">#4</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/5">#5</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/6">#6</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/7">#7</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/8">#8</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/9">#9</NavLink>
      </li>
      <li>
        <NavLink className="link" to="/10">#10</NavLink>
      </li>
    </ul>
  </nav>
);
export default Navigation;
