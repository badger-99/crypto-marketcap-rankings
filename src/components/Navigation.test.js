import { MemoryRouter } from "react-router";
import { render } from "@testing-library/react";
import Navigation from "./Navigation";

test('Navigation component renders as expected', () => {
  const navigation = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  )

  expect(navigation).toMatchSnapshot();
})