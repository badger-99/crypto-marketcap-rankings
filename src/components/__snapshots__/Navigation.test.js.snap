// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navigation component Navigation component renders loading spinner during api call 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
        id="loading"
      >
        <span
          id="loader"
          style="width: 8rem; height: 8rem; border-radius: 100%; border: 2px solid; border-top-color: #5788e4; border-bottom-color: transparent; border-left-color: #5788e4; border-right-color: #5788e4; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both; border-width: 7px;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
      id="loading"
    >
      <span
        id="loader"
        style="width: 8rem; height: 8rem; border-radius: 100%; border: 2px solid; border-top-color: #5788e4; border-bottom-color: transparent; border-left-color: #5788e4; border-right-color: #5788e4; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both; border-width: 7px;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
